ctr/tests/ctr32/big_endian.rs

```diff
@@ -0,0 +1,80 @@
+//! Counter Mode with a 32-bit big endian counter
+
+use hex_literal::hex;
+use stream_cipher::{NewStreamCipher, SyncStreamCipher};
+
+type Aes128Ctr = ctr::Ctr32BE<aes::Aes128>;
+
+const KEY: &[u8; 16] = &hex!("000102030405060708090A0B0C0D0E0F");
+const NONCE1: &[u8; 16] = &hex!("11111111111111111111111111111111");
+const NONCE2: &[u8; 16] = &hex!("222222222222222222222222FFFFFFFE");
+
+#[test]
+fn counter_incr() {
+    let mut ctr = Aes128Ctr::new(KEY.into(), NONCE1.into());
+    assert_eq!(ctr.current_ctr(), 0);
+
+    let mut buffer = [0u8; 64];
+    ctr.apply_keystream(&mut buffer);
+
+    assert_eq!(ctr.current_ctr(), 4);
+    assert_eq!(
+        &buffer[..],
+        &hex!(
+            "35D14E6D3E3A279CF01E343E34E7DED36EEADB04F42E2251AB4377F257856DBA
+                 0AB37657B9C2AA09762E518FC9395D5304E96C34CCD2F0A95CDE7321852D90C0"
+        )[..]
+    );
+}
+
+#[test]
+fn counter_seek() {
+    let mut ctr = Aes128Ctr::new(KEY.into(), NONCE1.into());
+    ctr.seek_ctr(1);
+    assert_eq!(ctr.current_ctr(), 1);
+
+    let mut buffer = [0u8; 64];
+    ctr.apply_keystream(&mut buffer);
+
+    assert_eq!(ctr.current_ctr(), 5);
+    assert_eq!(
+        &buffer[..],
+        &hex!(
+            "6EEADB04F42E2251AB4377F257856DBA0AB37657B9C2AA09762E518FC9395D53
+                 04E96C34CCD2F0A95CDE7321852D90C0F7441EAB3811A03FDBD162AEC402F5AA"
```

Comment:
Personally I simply disable rustfmt in such cases.

Reply:
Yeah, we can also look at extracting these out into blobby and turning these into proper stream cipher tests, although I still don't understand how all that works.